hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/parse/WrappedSqlParseException.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.parse;
+
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.impl.ParseException;
+
+/**
+ * An exception class which acts as a wrapper for Calcite SqlParseException.
+ */
+class WrappedSqlParseException extends SqlParseException  {
```

Comment:
I would avoid new exceptions because they complicate the debugging, but do not add much value. Our goal is to watch for a very specific situation - when we have a `SqlParseException` with a `ParseException` as a cause. This combination can only be thrown during the parsing. Therefore, we may extend the `QueryParser.parse` catch block, so that it alters the message that we add to the exception that we throw:
```
public QueryParseResult parse(String sql) {
    try {
        ...
    } catch (Exception e) {
        String message;
        if (<e is a SqlParserException with ParserException inside>)
            message = <trimmed message>
        } else {
            message = e.getMessage();
        }
        throw QueryException.error(SqlErrorCode.PARSING, message, e);
    }
```